services/motion/client.go

```diff
@@ -17,22 +17,23 @@ import (
 
 // client implements MotionServiceClient.
 type client struct {
+	resource.Named
+	resource.TriviallyReconfigurable
 	name   string
-	conn   rpc.ClientConn
 	client pb.MotionServiceClient
 	logger golog.Logger
 }
 
 // NewClientFromConn constructs a new Client from connection passed in.
-func NewClientFromConn(ctx context.Context, conn rpc.ClientConn, name string, logger golog.Logger) Service {
+func NewClientFromConn(ctx context.Context, conn rpc.ClientConn, name resource.Name, logger golog.Logger) (Service, error) {
 	grpcClient := pb.NewMotionServiceClient(conn)
 	c := &client{
-		name:   name,
-		conn:   conn,
+		Named:  name.AsNamed(),
+		name:   name.ShortNameForClient(),
```

Comment:
because to the person using the client, there's a fully qualified name with a remote. the client itself to send grpc must send the short name (without the remote). Named is public because all resources must be Named now.

Reply:
its also so it's cached

Reply:
see https://viam.atlassian.net/browse/RSDK-2728

Reply:
Thanks for making this ticket!